lib/pc_patch.c

```diff
@@ -522,6 +522,61 @@ pc_patch_from_patchlist(PCPATCH **palist, int numpatches)
 	return (PCPATCH*)paout;
 }
 
+// first: the first element to select (1-based indexing)
+// count: the number of points to select
+PCPATCH *
+pc_patch_range(const PCPATCH *pa, int first, int count)
+{
+	PCPATCH_UNCOMPRESSED *paout, *pu;
+	uint8_t *buf;
+	size_t size;
+	size_t start;
+
+	assert(pa);
+
```

Comment:
You could do first `--first;` up front to go from 1-based to 0-based indexing in order to simplify the indexing : no `-1` line 566 and 537-541 could become :

```
  --first;
  int countmax = pa->npoints - first;
  if ( count > countmax ) count = countmax ;
  if ( first < 0 || count <= 0) return NULL;
```